tests/checkpointing/test_model_checkpoint.py

```diff
@@ -1020,3 +1021,35 @@ def __init__(self, hparams):
     else:
         # make sure it's not AttributeDict
         assert type(ckpt[model.CHECKPOINT_HYPER_PARAMS_KEY]) == hparams_type
+
+
+def test_model_checkpoint_with_training_epoch_end(tmpdir):
+
+    """
+    This test assert ModelCheckpoint a warming is issued when monitor metric is used in training_epoch_end
+    """
+    class TestedModel(BoringModel):
+
+        def training_step(self, batch, batch_idx):
+            output = self.layer(batch)
+            loss = self.loss(batch, output)
+            self.log('train_loss', loss)
+            return {"loss": loss}
+
+        def training_epoch_end(self, outputs) -> None:
+            avg_loss = torch.stack([x["loss"] for x in outputs]).mean()
+            self.log('epoch_end_train_loss', avg_loss)
+            self.log('gb_step', self.global_step)
+
+    model = TestedModel()
+
+    chk = ModelCheckpoint(monitor='epoch_end_train_loss', save_top_k=-1)
+    with pytest.warns(UserWarning, match="Running first epoch, a MisconfigurationException"):
+        trainer = pl.Trainer(
+            default_root_dir=tmpdir,
+            max_epochs=3,
+            progress_bar_refresh_rate=1,
+            callbacks=[chk],
+        )
+        trainer.current_epoch = 2
+        trainer.fit(model)
```

Comment:
can we have some assertions to check that ModelCheckpoint did actually monitor as expected?

Reply:
Yes.